test/ProgressOnderwijsUtils.Analyzers.Test/DiagnosticHelper.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using ProgressOnderwijsUtils.Collections;
+
+namespace ProgressOnderwijsUtils.Analyzers.Test
+{
+    public static class DiagnosticHelper
+    {
+        public static Diagnostic[] GetDiagnostics(DiagnosticAnalyzer analyzer, string source)
+        {
+            var project = CreateProject(source);
+            var compilation = project.GetCompilationAsync().Result;
+            var compilationWithAnalyzers = compilation.WithAnalyzers(ImmutableArray.Create(analyzer));
+            return compilationWithAnalyzers.GetAllDiagnosticsAsync().Result.ToArray();
+        }
+
+        static Project CreateProject(string source)
+        {
+            var assemblyPath = Path.GetDirectoryName(typeof(object).Assembly.Location);
+
+            var projectId = ProjectId.CreateNewId();
+            var solution = new AdhocWorkspace()
+                .CurrentSolution
+                .AddProject(projectId, "TestProject", "TestProject", LanguageNames.CSharp)
+                .WithProjectCompilationOptions(projectId, new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary))
+                .AddMetadataReference(projectId, MetadataReference.CreateFromFile(Path.Combine(assemblyPath, "System.Runtime.dll")))
+                .AddMetadataReference(projectId, MetadataReference.CreateFromFile(typeof(object).Assembly.Location))
+                .AddMetadataReference(projectId, MetadataReference.CreateFromFile(typeof(Enumerable).Assembly.Location))
+                .AddMetadataReference(projectId, MetadataReference.CreateFromFile(typeof(Maybe).Assembly.Location));
+            var documentId = DocumentId.CreateNewId(projectId);
+            solution = solution.AddDocument(documentId, "Test.cs", SourceText.From(source));
+            return solution.GetProject(projectId);

```

Comment:
Kan me voorstellen dat het makkelijker is dit test project ook echt te maken, ipv hier via de api te defineren?

Reply:
Waarom dan? Nu staat het toch lekker lokaal en in memory voor alleen de testen?

Reply:
Naja, het is prima, maar gewoon niet super makkelijk te volgen; al helemaal dat System.Runtime ding, maar als het alternatief niet voor de hand liggend is (en ik kan zo 123 niks beters googlen met die AdhocWorkspace: 👍